### PR TITLE
Refactor ['GET'] Product from Wishlist

### PR DESCRIPTION
**Pull request by lombardero**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@
 Flask==1.1.1
 Flask-API==1.1
 Flask-SQLAlchemy==2.4.1
+flask-restplus==0.13.0
```
Comment (nisargthakkar) [inline requirements.txt:5]: this has already been defined. we can remove this

File: service/service.py [hunk 2]
```diff
@@ -108,6 +106,41 @@ def internal_server_error(error):
                    error='Internal Server Error',
                    message=message), status.HTTP_500_INTERNAL_SERVER_ERROR
 
+######################################################################
```
Comment (nisargthakkar) [inline service/service.py:101]: can we use the earlier defined Flask-RESTPlus `api` objects? It was tailored to the assignment's requirements.

File: service/service.py [hunk 3]
```diff
@@ -108,6 +106,41 @@ def internal_server_error(error):
...
+# Configure Swagger
+######################################################################
+api = Api(app,
+          version='1.0.0',
+          title='Wishlists REST API Service',
+          description='This is the Wishlists service.',
+          default='wihslists',
+          default_label='Wishlist shop operations',
+          doc='/apidocs', 
+          authorizations=authorizations
```
Comment (nisargthakkar) [inline service/service.py]: we don't need authorizations for the application

File: service/service.py [hunk 4]
```diff
@@ -219,22 +254,40 @@ def rename_wishlist(wishlist_id):
     return make_response(jsonify(wishlist.serialize()), status.HTTP_200_OK)
 
 ######################################################################
-# READ AN EXISTING ITEM FROM WISHLIST
+# PATH: /wishlists{id}/items/{id}
```
Comment (nisargthakkar) [inline service/service.py]: the path should be `/wishlists/{id}/items/{id}`

File: service/service.py [hunk 5]
```diff
@@ -219,22 +254,40 @@ def rename_wishlist(wishlist_id):
...
-    wishlist_product = WishlistProduct.find(wishlist_id, product_id)
-    if not wishlist_product:
-        raise NotFound("The wishlist-producttuple ({},{}) you are looking\
-                       for was not found.".format(wishlist_id, product_id))
-    return make_response(jsonify(wishlist_product.serialize()), status.HTTP_200_OK)
+        wishlist_product = WishlistProduct.find(wishlist_id, product_id)
+        if not wishlist_product:
+            api.abort(status.HTTP_404_NOT_FOUND, "The wishlist-product tuple ({},{}) you\
+                      are looking for was not found.".format(wishlist_id, product_id))
+            # raise NotFound("The wishlist-producttuple ({},{}) you are looking\
```
Comment (nisargthakkar) [inline service/service.py]: please remove the commented code

File: service/service.py [hunk 6]
```diff
@@ -32,28 +44,14 @@
...
-wishlist_model = api.model('Wishlist', {
-    'id': fields.Integer(readOnly=True,
-                         required=False,
-                         description='The unique id assigned internally by service'),
-    'name': fields.String(required=True,
-                          description='The name of the Wishlist'),
-    'customer_id': fields.Integer(required=True,
-                                  description='The id of the customer that owns the wishlist')
-})
+#Autorizations for Swagger docs
```
Comment (nisargthakkar) [inline service/service.py]: we don't need authorizations for our app

File: service/service.py [hunk 7]
```diff
@@ -153,15 +186,17 @@ def post(self):
         # location_url = api.url_for(WishlistResource, wishlist_id=wishlist.id, _external=True)
         location_url = '%s/wishlists/%s' % (request.base_url, wishlist.id)
         return message, status.HTTP_201_CREATED, { 'Location': location_url }
-
+        
 ######################################################################
 # GET INDEX
 ######################################################################
+@app.route('/')
```
Comment (nisargthakkar) [inline service/service.py]: does this work? Does the home page also show up at `/`?

**Review by nisargthakkar — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 1 above)
Comment (lombardero) [reply requirements.txt:5]: True, was due to the merge conflict.

File: service/service.py (hunk 3 above)
Comment (lombardero) [reply service/service.py]: Ok, removed.

File: service/service.py (hunk 2 above)
Comment (lombardero) [reply service/service.py:101]: I do not get what are you referring to in this comment.

File: service/service.py (hunk 4 above)
Comment (lombardero) [reply service/service.py]: It was a typo! Corrected!

File: service/service.py (hunk 6 above)
Comment (lombardero) [reply service/service.py]: Removed

File: service/service.py (hunk 5 above)
Comment (lombardero) [reply service/service.py]: Done

File: service/service.py (hunk 7 above)
Comment (lombardero) [reply service/service.py]: Nope, removed.

File: requirements.txt (hunk 1 above)
Comment (nisargthakkar) [reply requirements.txt:5]: yep. Thanks

File: service/service.py (hunk 2 above)
Comment (nisargthakkar) [reply service/service.py:101]: Our code contained Flask-RESTPlus instantiated before: https://github.com/nyu-devops-fall19-wishlists/wishlists/pull/182/files#diff-8bf0810d7919804091b3399999dc4b5fL38

Probably got lost due to merge conflicts.

File: service/service.py (hunk 3 above)
Comment (nisargthakkar) [reply service/service.py]: thanks

File: service/service.py (hunk 2 above)
Comment (lombardero) [reply service/service.py:101]: Yes. Changed back to what it was.

**Review by nisargthakkar — APPROVED** [review]
LGTM

**Pull request merged**